lib/objects/objectsUtils.js

```diff
@@ -332,13 +333,13 @@ function getDefaultAdminRights(acl, _isState) {
 
 function getUserGroup(objects, user, callback) {
     if (!user || typeof user !== 'string' || !user.match(regUser)) {
-        console.log('invalid user name: ' + user);
+        console.log(`invalid user name: ${user}`);
         user = JSON.stringify(user);
         // deep copy
-        return callback.call(objects, 'invalid user name: ' + user, user, [], deepClone(defaultAcl.acl));
+        return tools.maybeCallbackWithError(callback, objects, `invalid user name: ${user}`, user, [], deepClone(defaultAcl.acl));
```

Comment:
`objects` is the type of `this`, not the second argument to `maybeCallbackWithError` which should be an Error or null

Reply:
Question is: Why did this need `this` to be set to the objects instance?

Reply:
idk, then in the cb on object level it is explicitly set back: https://github.com/ioBroker/ioBroker.js-controller/blob/11df1b78ffcbca5052076457619398b7e9b95b7d/lib/objects/objectsInMemFileDB.js#L257

Reply:
from my point of view it makes no sense, because the cb is used in objects db where the this content is the objects db anyways.. @Apollon77 ?

Reply:
In fact with this you set the "scope of this" for the callback method, so the callbacks can do "this.objectsMethod" ... else they can not because this is pointing to the callback method.

Reply:
(my asumption) ...

Reply:
hm it is only used in the objects db where the scope is anyway set to the objects db.. problem of undocumented code..

Reply:
yes but here it is the "Objects Tools" ... so it would point to the incorrect instance

Reply:
or?!

Reply:
No the objects db code is calling the method with its own this context, as the objects argument, which perfectly makes sense, but then it's explicity set back for the callback where this is not needed:

https://github.com/ioBroker/db-objects-redis/blob/a266b8ff4196c73ef596627625aa282780488fe7/lib/objects/objectsInRedis.js#L624

Reply:
lets risk it ... we will see in testing if it has unawaited consequences